samples/dotnet/KernelBuilder/Program.cs

```diff
@@ -37,7 +40,13 @@
 // Manually setup all the dependencies used internally by the kernel
 var logger = NullLogger.Instance;
 var memoryStorage = new VolatileMemoryStore();
-var embeddingGenerator = new AzureTextEmbeddings("modelId", "https://...", "apiKey", "2022-12-01", logger);
+
+var httpClient = new HttpClient();
```

Comment:
Could/should this use `CreateAzureOpenAIServiceClient` instead?

Reply:
Ah I see it's private right now. Thoughts on pros/cons of exposing those for users?

Reply:
To be honest, looking at these classes a second time, I don't think any of them should be exposed to users. I think, users should only know about the IEmbeddingGenerator and ITextCompletionClient interfaces and a factory that can instantiate implementations of these interfaces. That should be it, otherwise it'll lead to breaking changes every time we change one of the implementations. Let' me think about it a little bit.